doc/features/open-telemetry/tracing-rfc.md

```diff
@@ -0,0 +1,413 @@
+
+```csharp
+using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+ .AddSource(CassandraInstrumentation.ActivitySourceName)
+ .AddConsoleExporter()
+ .Build();
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Cassandra.OpenTelemetry project
+
+### Dependencies and target frameworks
+
+Similar to the existent metrics feature, this functionality will include a project named `Cassandra.OpenTelemetry` that will extend the core `Cassandra` project and will handle the spans'generation.\
+`Cassandra.OpenTelemetry` has a single dependency from the package `System.Diagnostics.DiagnosticSource` which has the lowest non-deprecated version as [`6.0.1`](https://www.nuget.org/packages/System.Diagnostics.DiagnosticSource/6.0.1#dependencies-body-tab). Using this version it is possible to target the same .NET frameworks as the `Cassandra.AppMetrics` project which are `netstandard2.0;net461`.
+
+**Note:** There is an alternative that uses `OpenTelemetry.Api` package to avoid code duplication, but that implies changing the minimal target framework from `net461` to `net462`. Such alternative and its drawbacks are mentioned in the section [*Rationale and Alternatives*](#rationale-and-alternatives).
+
+### Extension methods
+
+The project will include a `Builder` extension method named `AddOpenTelemetryTraceInstrumentation` that will instantiate a new class named `Trace` which will start and populate the `System.Diagnostics.Activity` class that will have the Cassandra telemetry information.
+This new class will implement an `IDriverTrace` interface that will be included in the Cassandra core, which is a similar implementation that already exists in the `Cassandra.AppMetrics` package.
+
+## Cassandra core project
+
+### Tracer observers
+
+The tracing implementation will use the already defined factory interfaces for the creation of observer factories, so the following classes will be created:
+
+```csharp
+internal class TracerObserverFactoryBuilder : IObserverFactoryBuilder
+```
+
+```csharp
+internal class TracerObserverFactory : IObserverFactory
+```
+
+When instantiating the factories that will culminate in the creation if the `TracesRequestObserver`, the implementation of `IDriverTrace` will be passed as parameter. This interface will define the following methods:
+
+```csharp
+public interface IDriverTrace
+{
+    void OnStart(IStatement statement);
+
+    void OnError(Exception ex);
+}
+```
+
+and will be used in the `TracesRequestObserver`:
+
+```csharp
+internal class TracesRequestObserver : IRequestObserver
+
+(...)
+
+public void OnRequestError(Host host, RequestErrorType errorType, RetryDecision.RetryDecisionType decision)
+{
+}
+
+public void OnRequestFinish(Exception exception)
+{
+    _driverTrace.OnError(exception);
+}
+
+public void OnRequestStart(IStatement statement)
+{
+    _driverTrace.OnStart(statement);
+}
+
+public void OnSpeculativeExecution(Host host, long delay)
+{
+}
+```
```

Comment:
Ok I'm going a bit back and forth on this (see this [discussion](https://github.com/datastax/csharp-driver/pull/603#discussion_r1478666554)) but now I realize we pretty much have to add node level tracing now otherwise we will need breaking changes to the interface in the future to add it.

Reply:
I'm gonna close this discussion so we can focus on the `IRequestTracker` [discussion](https://github.com/datastax/csharp-driver/pull/603#discussion_r1478741930) to avoid going back and forth :)